lib/TimeZoneUtil.php

```diff
@@ -14,94 +24,39 @@
  */
 class TimeZoneUtil
 {
-    public static $map = null;
+    private static $instance = null;
 
-    /**
-     * List of microsoft exchange timezone ids.
-     *
-     * Source: http://msdn.microsoft.com/en-us/library/aa563018(loband).aspx
-     */
-    public static $microsoftExchangeMap = [
```

Comment:
I am not sure we can drop this field, as other code might rely on it beeing it.

there is more public api changed with this PR in a non-BC manner

Reply:
Right!. I'll readd those other public items to stay backwards compatible.

Main question though is whether I shall do it in a way that allows them to be easily removed at a later stage (meaning to duplicate some code) or to have as little duplicate code as possible.

Personally I'd opt for the former version but am happy to modify it to the later one.